### ISSUE TITLE
ES bindings for constants on Globals

### ISSUE THREAD

**Issue by tobie** [heycam/webidl#528]
The [spec says](https://heycam.github.io/webidl/#es-constants):

> The location of the property is determined as follows:
> 1. If the interface was declared with the [Global] extended attribute, then the property exists on the single object that implements the interface.
> 2. Otherwise, if the interface has an interface prototype object, then the property exists on it.

In practice, that's not what's implemented by browsers, however.

The only constants I'm aware of on globals are: `Window.TEMPORARY` and `Window.PERSISTENT` (thanks, @TimothyGu, for the heads up). And that's only exposed in Chrome. Maybe there are others?

In Chrome, the property is exposed on the interface prototype object, and so there is no special casing even though it's a global.

Proposed resolution here is simply to remove the special casing for [Global].

**Comment by bzbarsky** [comment]
There's a reason globals are special-cased in terms of property location.  It's because putting things on the prototype object breaks horribly when people try to polyfill and the var shadows the actual property.  That's why methods and attributes are special-cased in terms of location.

Maybe this is not a problem for constants, but it's hard to tell without actual use cases when people try to make their code work both in browsers with and without the constants.

**Comment by domenic** [comment]
My proposed resolution is to disallow constants on globals, as in general we shouldn't be using constants very much, and almost certainly not on the global object.

We could always revisit if someone has a compelling use case, but that's my instinct for now.

**Comment by bzbarsky** [comment]
That would work for me.

**Comment by Ms2ger** [comment]
@domenic is Chrome planning to remove the TEMPORARY/PERSISTENT constants from Window?

**Comment by domenic** [comment]
I'm not aware of any work to do so, although I will file a bug. Regardless, I don't think such nonstandard constants should impact our decisions in spec-space.

**Comment by domenic** [comment]
Filed https://bugs.chromium.org/p/chromium/issues/detail?id=991584. This is part of the webkitFileSystem API so I'm unsure if it will get removed before that whole API gets removed.

**Comment by inexorabletash** [comment]
Yeah, the constants are used at the API entry point (for this and a legacy quota API). Usage is still high so removal is going to be a long slog.
